sycl/test-e2e/Compression/compression.cpp

```diff
@@ -0,0 +1,24 @@
+// End-to-End test for testing device image compression.
```

Comment:
I would also add a test case where you have multiple translation units defining different kernels where some of those translation units are built with compression enabled and some of them were not.

NOTE: considering that compression happens at link time, you need to read "translation unit" in the example above as "fully linked library". I'm not sure if we invoke `clang-linker-wrapper` when we build static libraries, but we for sure do so for shared libraries.